api/controllers/OpenEAppController.js

```diff
@@ -64,44 +62,24 @@ const OpenEAppController = {
     },
 
     _getApplicationDataFromCasebook(req, res) {
-        const { hmacKey, customURLs, casebookCertificate, casebookKey } =
-            req._sails.config;
+        const url = req._sails.config.customURLs.applicationStatusAPIURL;
         const queryParamsObj = {
             timestamp: new Date().getTime().toString(),
             applicationReference: req.params.unique_app_id,
         };
-        const queryParams = new URLSearchParams(queryParamsObj);
-        const queryStr = queryParams.toString();
-        const hash = crypto
-            .createHmac('sha512', hmacKey)
-            .update(Buffer.from(queryStr, 'utf-8'))
-            .digest('hex')
-            .toUpperCase();
-        const options = {
-            uri: customURLs.applicationStatusAPIURL,
-            agentOptions: {
-                cert: casebookCertificate,
-                key: casebookKey,
-            },
-            method: 'GET',
-            headers: {
-                hash,
-                'Content-Type': 'application/json; charset=utf-8',
-            },
-            json: true,
-            qs: queryParamsObj,
-        };
 
-        return request(options)
+        return CasebookService.get(url, {
+            params: queryParamsObj,
+        })
             .then((response) => {
                 const isErrorResponse =
-                    typeof response === 'object' &&
-                    response.hasOwnProperty('errors');
+                    typeof response.data === 'object' &&
+                    response.data.hasOwnProperty('errors');
                 if (isErrorResponse) {
-                    sails.log.error(response.message);
+                    sails.log.error(response.data.message);
                     return res.serverError();
                 }
-                return response;
+                return response.data;
```

Comment:
would it just be easier to check for response code?

Reply:
Good point. For some reason, I thought that casebook would return a 200 but contain an error with a message but I just check the documentation they sent and that would only happen for 400's or 422's. I'll get that sorted.